front/src/hooks/useDeleteBoxes.ts

```diff
@@ -0,0 +1,140 @@
+import { gql, useMutation } from "@apollo/client";
+import { useCallback, useState } from "react";
+import { IBoxBasicFields } from "types/graphql-local-only";
+import { useErrorHandling } from "./useErrorHandling";
+import { useNotification } from "./useNotification";
+
+export enum IDeleteBoxResultKind {
+  SUCCESS = "success",
+  FAIL = "fail",
+  NETWORK_FAIL = "networkFail",
+  NOT_AUTHORIZED = "notAuthorized",
+  INVALID_IDENTIFIERS = "invalidIdentifiers",
+}
+
+export interface IDeleteBoxResult {
+  kind: IDeleteBoxResultKind;
+  requestedBoxes: IBoxBasicFields[];
+  deletedBoxes?: IBoxBasicFields[];
+  invalidIdentifiers?: string[];
+  error?: any;
+}
+
+export const DELETE_BOXES = gql`
+  mutation DeleteBoxes($labelIdentifiers: [String!]!) {
+    deleteBoxes(labelIdentifiers: $labelIdentifiers) {
+      ... on BoxResult {
+        updatedBoxes {
+          deletedOn
+          history {
+            changes
+          }
+        }
+        invalidBoxLabelIdentifiers
+      }
+    }
+  }
+`;
+
+export const useDeleteBoxes = () => {
+  const { triggerError } = useErrorHandling();
+  const { createToast } = useNotification();
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const [deleteBoxesMutation] = useMutation(DELETE_BOXES);
+
+  const deleteBoxes = useCallback(
+    (boxes: IBoxBasicFields[], showToasts: boolean = true, showErrors: boolean = true) => {
+      setIsLoading(true);
+      const labelIdentifiers = boxes.map((box) => box.labelIdentifier);
+
+      return deleteBoxesMutation({
+        variables: {
+          labelIdentifiers,
+        },
+      })
+        .then(({ data, errors }) => {
+          setIsLoading(false);
+
+          if (errors?.length) {
+            const errorCode = errors[0]?.extensions?.code;
+
+            if (errorCode === "FORBIDDEN") {
```

Comment:
For permission errors, the mutations won't return FORBIDDEN error code but an InsufficientPermissionError as part of the DeleteBoxesResult union.